### DISCUSSION TITLE
Feature Request: Variable Naming

### DISCUSSION THREAD

**Discussion by eliorc** [community/community#40524] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Every programmer has experienced the agony of not knowing how to name his/her variables.

Github Copilot by default suggests the trailing code for completion, but knowing how language models are built it is just as simple to suggest leading code for completion.

It would be great if it was possible to write something like
```python3
*** = [x for x in range(10) if x % 2 == 0]
```
and get a suggestion like to replace `***` with `even_numbers`.

There might be other cases for leading code completion, but naming variables is a real thing

**Comment by labeleddata** [comment]
This is a very pragmatic and elegant application of AI to a real human problem. The idea is really nice. Implementation would be awesome as well

**Reply by ggoonnzzaallo** [reply]
@labeleddata I ended up building my own extension for variable naming since Microsoft ignored this issue. If interested to try it, would love to get your feedback, can find it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot)
![Screenshot 2024-02-28 at 4 56 22 PM](https://github.com/community/community/assets/121972034/0dbf5e23-53e1-4f7b-bca4-1b999deb51ec)

**Reply by tendiesfortwo** [reply]
@ggoonnzzaallo looks neat, will check it out

**Comment by iamcjwilliams** [comment]
@eliorc Thank you for your suggestion! 🚀 I'll be sure to notify our copilot team!

**Comment by ggoonnzzaallo** [comment]
It's been 1 year+ and this is still not something Copilot can handle...

I actually end up building my own Copilot extension that handles variable naming, you can try it here [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot). 

There are only two hard things in Computer Science: cache invalidation and naming things. This extension solves one of them :)
